core/providers/api/provider.tsx

```diff
@@ -1,30 +1,50 @@
-import React, { useEffect, useReducer } from 'react'
-import { ApiPromise, WsProvider } from '@polkadot/api'
-import { useChains } from '../../hooks/mod.ts'
-import { APIContext } from './context.ts'
-import { apiProvidersReducer } from './reducer.ts'
-import { useConfig } from '../../mod.ts'
+import React, { useEffect, useReducer } from "react";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { useChains } from "../../hooks/mod.ts";
+import { APIContext } from "./context.ts";
+import { apiProvidersReducer } from "./reducer.ts";
+import { useConfig } from "../../mod.ts";
+import { Sc, ScProvider } from "../../../utils/mod.ts";
+import { Chain } from "../../../chains/mod.ts";
 
 export const APIProvider: React.FC<React.PropsWithChildren<any>> = ({
   children,
 }) => {
-  const chains = useChains()
-  const { chainRpcs } = useConfig()
-  const [apis, dispatch] = useReducer(apiProvidersReducer, {})
+  const chains = useChains();
+  const { chainRpcs } = useConfig();
+  const [apis, dispatch] = useReducer(apiProvidersReducer, {});
+
+  const addApiProvider = (chain: Chain) => {
+    const provider = new WsProvider(chainRpcs[chain.id] || chain.rpcs[0]);
+
+    ApiPromise.create({ provider }).then((api) => {
+      dispatch({
+        type: "ADD_API_PROVIDER",
+        chainId: chain.id,
+        apiProvider: { api, provider },
+      });
+    });
+  };
+
+  const addSmoldotProvider = async (chain: Chain) => {
```

Comment:
We should only set one provider at a time, not both the light client and a WsProvider instance.

I think it makes sense to add a config in the ConfigProvider so a user or the developer can choose if they want one or the other globally or per chain. The default should be WsProvider.

1. Maybe something like this?

// in config/model.ts

```ts
export type ProviderType = 'trusted' | 'light-client';

export type ConfigProps = {
  // ... omitted
  api?: {
    default?: ProviderType;
  } & Partial<Record<ChainId, ProviderType>>;
};
```

Then inside of ApiProvider you use `useConfig()` to get the values and create the instances accordingly.

Reply:
thx, made some changes, will continue with `ApiProvider` in the upcoming commits